charts/consul/templates/ingress-gateways-deployment.yaml

```
@@ -107,374 +146,172 @@ spec:
-                  {{- if $root.Values.global.enableConsulNamespaces }}
-                  namespace = "{{ (default $defaults.consulNamespace .consulNamespace) }}"
-                  {{- end }}
-                  {{- if $root.Values.global.adminPartitions.enabled }}
-                  partition = "{{ $root.Values.global.adminPartitions.name }}"
-                  {{- end }}
-                  port = ${WAN_PORT}
-                  address = "${WAN_ADDR}"
-                  tagged_addresses {
-                    lan {
-                      address = "${POD_IP}"
-                      port = 21000
-                    }
-                    wan {
-                      address = "${WAN_ADDR}"
-                      port = ${WAN_PORT}
-                    }
-                  }
-                  proxy {
-                    config {
-                      {{- if (and $root.Values.global.metrics.enabled $root.Values.global.metrics.enableGatewayMetrics) }}
-                      envoy_prometheus_bind_addr = "${POD_IP}:20200"
-                      {{- end }}
-                      envoy_gateway_no_default_bind = true
-                      envoy_gateway_bind_addresses {
-                        all-interfaces {
-                          address = "0.0.0.0"
-                        }
-                      }
-                    }
-                  }
-                  checks = [
-                    {
-                      name = "Ingress Gateway Listening"
-                      interval = "10s"
-                      tcp = "${POD_IP}:21000"
-                      deregister_critical_service_after = "6h"
-                    }
-                  ]
-                }
-                EOF
-
-                /consul-bin/consul services register \
-                  {{- if $root.Values.global.acls.manageSystemACLs }}
-                  -token-file=/consul/service/acl-token \
-                  {{- end }}
-                  /consul/service/service.hcl
-          volumeMounts:
-            - name: consul-service
-              mountPath: /consul/service
-            - name: consul-bin
-              mountPath: /consul-bin
-            {{- if $root.Values.global.tls.enabled }}
-            {{- if $root.Values.global.tls.enableAutoEncrypt }}
-            - name: consul-auto-encrypt-ca-cert
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        command:
+        - /bin/sh
+        - -ec
+        - |
+          consul-dataplane \
+            -envoy-ready-bind-address=$POD_IP \
+            -envoy-ready-bind-port=21000 \
+            {{- if $root.Values.externalServers.enabled }}
+            -addresses={{ $root.Values.externalServers.hosts | first }} \
```

we need to quote it so that command like `-addresses="exec=..."` works
```suggestion
            -addresses={{ $root.Values.externalServers.hosts | first | quote }} \
```